cloudify_vsphere/devices/__init__.py

```diff
@@ -136,6 +148,9 @@ def attach_server_to_ethernet_card(ctx, **kwargs):
 
 @operation
 def detach_controller(ctx, **kwargs):
+    if 'busKey' not in ctx.source.instance.runtime_properties:
+        ctx.logger.info("Contoller dettached.")
```

Comment:
*Controller detached

Also, this should perhaps say "Controller was not attached, skipping." or something similar to indicate that no action was necessary, and the reason why.